### PR TITLE
Refactored DIB-R and added textured DIB-R example

### PR DESCRIPTION
**Pull request by TommyX12**
Improved dib-r example (in `examples/renderers/DIB-R`) to be able to render textured mesh (#97). Refactored dib-r to be more intuitive for texturing.

Tested on Ubuntu 16.04, CUDA 10.0, PyTorch 1.3.1.

### CODE REVIEW

File: kaolin/graphics/dib_renderer/renderer/phongrender.py [hunk 1]
```diff
@@ -83,10 +86,10 @@ def forward(self, points, cameras, colors, lightdirect_bx3, material_bx3x3, shin
         bnum = normal1_bxfx3.shape[0]
 
         # we have uv, normal, eye to interpolate
-        uv_bxpx2, ft_fx3, texture_bx3xthxtw = colors
-        c0 = uv_bxpx2[:, ft_fx3[:, 0], :]
-        c1 = uv_bxpx2[:, ft_fx3[:, 1], :]
-        c2 = uv_bxpx2[:, ft_fx3[:, 2], :]
+        # uv_bxpx2, texture_bx3xthxtw = colors
```
Comment (Caenorst) [inline kaolin/graphics/dib_renderer/renderer/phongrender.py]: Please remove instead of comment it out

**Review by Caenorst — COMMENTED** [review]
Can you specify the tests that you ran (and with what system) ? I see a bunch of modifications in the API of core functions, they seems reasonable but I'm afraid it could break something.

**Comment by TommyX12** [comment]
> Can you specify the tests that you ran (and with what system) ? I see a bunch of modifications in the API of core functions, they seems reasonable but I'm afraid it could break something.

I believe there's currently no test in place for dib-r yet, and no other module is using the core API, so the change shouldn't affect anything.
I have also added the system info in the description (Ubuntu 16.04, CUDA 10.0, PyTorch 1.3.1).

File: kaolin/graphics/dib_renderer/renderer/phongrender.py (hunk 1 above)
Comment (TommyX12) [reply kaolin/graphics/dib_renderer/renderer/phongrender.py]: Resolved in latest commit.

**Review by Caenorst — APPROVED** [review]
LGTM

**Comment by xiangdonglai** [comment]
Is 'ft_fx3' no longer supported? I think it is quite useful.

**Comment by TommyX12** [comment]
> Is 'ft_fx3' no longer supported? I think it is quite useful.

`ft_fx3` is the face-to-vertex information. That information is already provided in the `points` argument, which should be a tuple of vertex position tensor and face information tensor. So it's not removed, it was just redundant. In the original DIB-R design, the same information needs to be passed to the method twice, so I removed one of them to reduce the chances of human error.

**Comment by xiangdonglai** [comment]
There might be situations where ft_fx3 is not the same as faces_fx3. Sometimes ft_fx3 stores the indices of vertex texture coordinates ('uv' also known as 'vt'), which is different from face_fx3 that stores indices of vertices.

For example, on this [wiki page for obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file), 'v1' are 'vt1' are different. v1,v2,v3 belong to faces_fx3, while vt1,vt2,vt3 belong to ft_fx3.
image

Besides, the util function that loads an obj file in kaolin also explicitly distinguishes between 'facenp_fx3' and 'ft_fx3'.
https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/graphics/dib_renderer/utils/mesh.py#L261

**Comment by TommyX12** [comment]
> There might be situations where ft_fx3 is not the same as faces_fx3. Sometimes ft_fx3 stores the indices of vertex texture coordinates ('uv' also known as 'vt'), which is different from face_fx3 that stores indices of vertices.
> 
> For example, on this [wiki page for obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file), 'v1' are 'vt1' are different. v1,v2,v3 belong to faces_fx3, while vt1,vt2,vt3 belong to ft_fx3.
> image
> 
> Besides, the util function that loads an obj file in kaolin also explicitly distinguishes between 'facenp_fx3' and 'ft_fx3'.
> https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/graphics/dib_renderer/utils/mesh.py#L261

That's a good point. I will add it back in a future PR.

**Comment by xiangdonglai** [comment]
> > There might be situations where ft_fx3 is not the same as faces_fx3. Sometimes ft_fx3 stores the indices of vertex texture coordinates ('uv' also known as 'vt'), which is different from face_fx3 that stores indices of vertices.
> > For example, on this [wiki page for obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file), 'v1' are 'vt1' are different. v1,v2,v3 belong to faces_fx3, while vt1,vt2,vt3 belong to ft_fx3.
> > image
> > Besides, the util function that loads an obj file in kaolin also explicitly distinguishes between 'facenp_fx3' and 'ft_fx3'.
> > https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/graphics/dib_renderer/utils/mesh.py#L261
> 
> That's a good point. I will add it back in a future PR.

That'll be great. Thanks.

**Pull request merged**